src/sql/src/names.rs

```diff
@@ -1761,20 +1762,52 @@ impl<'a> Fold<Raw, Aug> for NameResolver<'a> {
                 print_id: _,
             } => {
                 let item = self.catalog.get_item(id).at_version(*version);
-                let desc = match item.desc(full_name) {
-                    Ok(desc) => desc,
-                    Err(e) => {
-                        if self.status.is_ok() {
-                            self.status = Err(e.into());
+                let name = normalize::column_name(column_name.column.clone());
+                let desc = match item.item_type() {
+                    CatalogItemType::Type => {
+                        let details = item
+                            .type_details()
+                            .expect("type items must carry type details");
+                        match details.typ.desc(self.catalog) {
+                            Ok(Some(desc)) => Cow::Owned(desc),
+                            Ok(None) => {
+                                if self.status.is_ok() {
+                                    self.status = Err(PlanError::UnknownColumn {
+                                        table: Some(full_name.clone().into()),
+                                        column: name.clone(),
+                                        similar: Box::<[ColumnName]>::from([]),
+                                    });
```

Comment:
Is this the correct error type? Feels like instead of saying "this column doesn't exist" we should return "this type has no columns".